File: inbloom-hello-world-ruby/app/controllers/recipes/recipe11.rb
# Assign a Grade to a Student
def recipe11()
  grade_loc, grade = create_grade()
  
  RestClient.delete grade_loc, headers
  
  return grade_loc, grade
end

def create_grade()
  new_grade = grade(
    "2000-2001",
    "Final",
    "d0947e676f376eedc3775993a4e5a309d588e553_id",
    "19b6e33399c60f9b839959fbf9d1bea50aea4477_id",
    "f6683a18d00d897c59c1533170fe1fb3eb7e23d6_id",
    "d0947e676f376eedc3775993a4e5a309d588e553_id6319342cc147e3038ac0894a4fa24511c90d4656_id"
  )
  
  endpoint = REST_URL + "grades"
  response = RestClient.post endpoint, new_grade.to_json, headers
  grade_loc = response.headers[:location]  
  
  logger.info "new grader URL: " + grade_loc
  
  return grade_loc, new_grade
  
end

def grade( school_year, grade_type, section_id, student_id, grading_period_id, student_section_associations_id,  optional ={})
  grade_hash = {
    :studentSectionAssociationId => student_section_associations_id,
    :sectionId => section_id,
    :studentId => student_id,
    :gradingPeriodId => grading_period_id,
    :schoolYear => school_year,
    :gradeType => grade_type
  }
  grade_hash.update optional

end

File: hello-world-javascript/routes/recipes/recipe_11.js
/**
 * routes/recipes/recipe_11.js
 * Handler for Recipe #11 requests.
 *
 * @author  Lindsay Parker
 * @version 1.0.0, 06 JUN 2013
 */
 
 var config = require('../../config');
  
  
/**
 * Assign a grade to a student.
 * Recipe #11
 * @param client
 * @return function
 */
 exports.createGradeForStudent = function (client){
 	return function(req, res){

 	    var recipeDescription = 'Assign a grade to a student';

 		// Assign existing section ID, student ID, and student/section association ID for test purposes.
 		// These IDs will most likely be passed into the function in a functional application.
 		var sectionId = SECTION_ID_HERE;
 		var studentId = STUDENT_ID_HERE;
 		var studentSectionAssociationId = STUDENT_SECTION_ASSOC_ID_HERE;
 		
 		// Create path to 'grades' endpoint
		var path = config.api.base_url + config.api.api_call_append + config.api.api_version +
					'/grades';
		
		var grade = {
			'gradeType': 'Exam',
			'letterGradeEarned': 'A+',
			'numericGradeEarned': 99,
			'schoolYear': '2011-2012',
			'sectionId': sectionId,
			'studentId': studentId,
			'studentSectionAssociationId': studentSectionAssociationId
		};

		// POST request to api using Restify module .post function.
		// 'client' object is passed into function (Created in 'Hello World')
		client.post(path, grade, function (err, req2, res2, obj) {

			var newGradeEndpoint = res2.headers['location'];
				
			// Prints new Assessment endpoint to webpage for verification
			res.render('recipes', { tab_name: 'Recipes', 
									title: 'Recipe 11',
									description: recipeDescription,
									data: newGradeEndpoint }
			);
		});
 	};
 };
 